packages/node/src/soroban/soroban.server.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020-2022 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { getLogger } from '@subql/node-core';
+import { Server, SorobanRpc } from 'soroban-client';
+import { GetEventsRequest } from 'soroban-client/lib/server';
+
+const logger = getLogger('soroban-server');
+const DEFAULT_PAGE_SIZE = 100;
+
+export class SorobanServer extends Server {
+  private eventsCache: { [key: number]: SorobanRpc.GetEventsResponse } = {};
+
+  async getEvents(
+    request: GetEventsRequest,
+  ): Promise<SorobanRpc.GetEventsResponse> {
+    if (this.eventsCache[request.startLedger]) {
+      const cachedEvents = this.eventsCache[request.startLedger];
+      delete this.eventsCache[request.startLedger];
+      return new Promise((resolve) => {
+        //resolving immediately will make performance score go to NaN
```

Comment:
This is a problem with the performance benchmarking, we should fix the issue there.